server/routes/contacts.js

```diff
@@ -43,17 +43,22 @@ router.get("/:email/contacts", async (req, res) => {
       contactsList = approvedContacts
         .sort((a, b) => sortByEmail(a, b))
         .map((contact, index) => {
+          const isOnline = Object.keys(onlineUsers).includes(contact.email);
           return {
             email: contact.email,
             name: sortUsersByEmail[index].name,
             pictureUrl: sortUsersByEmail[index].pictureURL,
-            conversationID: contact.conversationID
+            conversationID: contact.conversationID,
+            isOnline: isOnline,
```

Comment:
I think online status shouldn't be returned with contact list because whenever user logs out whole contact list will be reloaded in frontend. You can probably change status of particular user using sockets.

Reply:
When user first logs in we need to show current on/offline status of contacts. For that purpose onlineUsers object list is used. 

When current contact's status changes updateContacts() is triggered that directly changes our context global state without making requests to backend knowing that onlineUsers list updated. As I understand when context global state changes the page is not reloaded.